sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorAsyncClient.java

```diff
@@ -0,0 +1,167 @@
+     * @param eventHubName The event hub name.
+     */
+    EventProcessorAsyncClient(EventHubAsyncClient eventHubAsyncClient, String consumerGroupName,
+        PartitionProcessorFactory partitionProcessorFactory, EventPosition initialEventPosition,
+        PartitionManager partitionManager,
+        String eventHubName) {
+        this.eventHubAsyncClient = Objects
+            .requireNonNull(eventHubAsyncClient, "eventHubAsyncClient cannot be null");
+        this.consumerGroupName = Objects
+            .requireNonNull(consumerGroupName, "consumerGroupname cannot be null");
+        this.partitionProcessorFactory = Objects
+            .requireNonNull(partitionProcessorFactory, "partitionProcessorFactory cannot be null");
+        this.partitionManager = Objects
+            .requireNonNull(partitionManager, "partitionManager cannot be null");
+        this.initialEventPosition = Objects
+            .requireNonNull(initialEventPosition, "initialEventPosition cannot be null");
+        this.instanceId = UUID.randomUUID().toString();
+        this.eventHubName = Objects
+            .requireNonNull(eventHubName, "eventHubName cannot be null");
+    }
+
+    /**
+     * Starts the event processor. This method is synchronized. Only one thread can run either {@link #start()} or {@link
+     * #stop()}.
+     */
+    public synchronized void start() {
+        if (!started.compareAndSet(false, true)) {
+            logger.info("Event processor is already running");
+            return;
+        }
+
+        scheduler = Schedulers.newElastic("EventProcessorAsyncClient");
+        runner = scheduler.schedule(this::run);
+    }
+
+    /**
+     * Stops the event processor. This method is synchronized. Only one thread can run either {@link #start()} or {@link
+     * #stop()}.
+     */
+    public synchronized void stop() {
+        if (!started.compareAndSet(true, false)) {
+            logger.info("Event processor has already stopped");
+            return;
+        }
+
+        this.partitionConsumers.forEach((key, value) -> {
+            try {
+                logger.info("Closing event hub consumer for partition {}", key);
+                value.close();
+                logger.info("Closed event hub consumer for partition {}", key);
+                partitionConsumers.remove(key);
+            } catch (IOException ex) {
+                logger.warning("Unable to close event hub consumer for partition {}", key);
+            }
+        });
+        runner.dispose();
+        scheduler.dispose();
+    }
+
+    /* Internal implementation. This is only a simple demo to show how it may look
+    - will need to move it to a different class and add more details to this */
+    void run() {
+        /* This should run periodically get new ownership details and close/open new
+        consumers when ownership of this instance has changed */
+
+        List<String> eventHubPartitionIds = new ArrayList<>();
```

Comment:
I'm not fond of the .blockLast() in this synchronous method because it'll throw an exception if we use a parallel or single scheduler... Since this runs periodically, I have to think more about a better way to do this.

Here's a way to reduce the number of .blocks you are using.

```java

final Flux<PartitionOwnership> ownershipFlux = partitionManager.listOwnership(eventHubName, consumerGroupName)
    .cache();

eventHubAsyncClient.getPartitionIds().flatMap(id -> {
    return ownershipFlux.filter(ownership -> id.equals(ownership.partitionId()))
        // Ownership has never been claimed, so it won't exist in the list, so we provide a default.
        .single(new PartitionOwnership().partitionId(id));
}).flatMap(ownershipInfo -> {
    return partitionManager.claimOwnership(Collections.singletonList(ownershipInfo)).doOnComplete(() -> {
        logger.info("Claimed ownership");
    }).doOnError(error -> {
        logger.error("Unable to claim ownership", error);
    });
}).blockLast(Duration.ofSeconds(10));
```